docs/modules/knowledge-graph/pages/index.adoc

```diff
@@ -0,0 +1,44 @@
+= Graph-based Knowledge Extraction and Traversal
+
+RAGStack offers two libraries supporting knowledge graph extraction and traversal, `ragstack-ai-knowledge-graph` and `ragstack-ai-knowledge-store`.
+
+A knowledge graph represents distinct entities (real world objects, like "Marie Curie" or "maize") as **nodes**. Nodes are connected by **edges** indicating relationships between them. Each edge includes the source (for example, "Marie Curie" the person), the target ("Nobel Prize" the award) and a type, indicating how the source relates to the target (for example, “won”).
+
+A graph database isn't required to use the knowledge graph libraries - RAGStack uses Astra DB or Apache Cassandra to store and retrieve graphs.
+
+*TL;DR:*
+
+. Retrieval augmented generation based on chunking and vector similarity search has some weaknesses.
+.. Similarity search looks for information most similar to the question. This makes is harder to answer questions with multiple topics.
+.. Similarity search limits the number of chunks retrieved. What if there is similar information spread across multiple places? A similarity search must choose between retrieving multiple copies of the information (inefficient), or choosing only one copy (less context).
+. Knowledge graphs address these shortcomings. For example, if multiple sources have similar information, that knowledge is stored as one node instead of as disparate chunks.
+. From a developer's perspective, a knowledge graph is built into a RAG pipeline similarly to a vector search. The difference is in the underlying data structure and how the information is stored and retrieved.
+
+== The `ragstack-ai-knowledge-graph` library
+
+The `ragstack-ai-knowledge-graph` library contains functions for the extraction and traversal of knowledge graphs.
+
+To install the library, run:
+
+[source,bash]
+====
+pip install ragstack-ai-knowledge-graph
+====
+
+For more information, see xref:knowledge-graph.adoc[].
+
+== The `ragstack-ai-knowledge-store` library
+
+The `ragstack-ai-knowledge-store` library contains functions for creating a hybrid vector-and-graph knowledge store. This store combines the benefits of vector search with the context and relationships of a knowledge graph by describing the relationships between Document chunks with edges.
```

Comment:
Alternative ways to present knowledge store:

1. It's a knowledge graph with a general purpose schema represent the content of your unstructured documents. Instead of identifying specific entities, the nodes correspond to things like "text passages" (chunks) and edges capture relationships between those passages such as "links to". This is much more scalable to construct, because you don't need a specific schema or an LLM to extract information, and it is easier to consume by an LLM.
2. It's a vector store to which we've added "related edges" or "see also" paths. This brings the benefits of a knowledge graph (providing relevant, non-semantic similar information) without requiring the extra work of extracting a detailed knowledge graph. It can (conceptually/theoretically/not yet implementedly) be "added" to a vector implementation.